internal/rpcserver/rpcwebsocket.go

```diff
@@ -2173,6 +2261,81 @@ func handleStopNotifyNewTransactions(_ context.Context, wsc *wsClient, _ interfa
 	return nil, nil
 }
 
+// handleNotifyMixMessages implements the notifymixmessages command extension
+// for websocket connections.
+func handleNotifyMixMessages(_ context.Context, wsc *wsClient, _ interface{}) (interface{}, error) {
+	wsc.rpcServer.ntfnMgr.RegisterMixMessages(wsc)
+	return nil, nil
+}
+
+// handleStopNotifyMixMessages implements the stopnotifymixmessages command
+// extension for websocket connections.
+func handleStopNotifyMixMessages(_ context.Context, wsc *wsClient, _ interface{}) (interface{}, error) {
+	wsc.rpcServer.ntfnMgr.UnregisterMixMessages(wsc)
+	return nil, nil
+}
+
+func handleSendMixMessage(_ context.Context, wsc *wsClient, icmd interface{}) (interface{}, error) {
+	c := icmd.(*types.SendMixMessageCmd)
+
+	// Allocate a message of the appropriate type based on the wire
+	// command string.
+	var msg mixing.Message
+	switch c.Command {
+	case wire.CmdMixPairReq:
+		msg = new(wire.MsgMixPairReq)
+	case wire.CmdMixKeyExchange:
+		msg = new(wire.MsgMixKeyExchange)
+	case wire.CmdMixCiphertexts:
+		msg = new(wire.MsgMixCiphertexts)
+	case wire.CmdMixSlotReserve:
+		msg = new(wire.MsgMixSlotReserve)
+	case wire.CmdMixDCNet:
+		msg = new(wire.MsgMixDCNet)
+	case wire.CmdMixConfirm:
+		msg = new(wire.MsgMixConfirm)
+	case wire.CmdMixFactoredPoly:
+		msg = new(wire.MsgMixFactoredPoly)
+	case wire.CmdMixSecrets:
+		msg = new(wire.MsgMixSecrets)
+	default:
+		err := rpcDeserializationError("Unrecognized mixing message "+
+			"wire command string %q", c.Command)
+		return nil, err
+	}
+
+	// Deserialize message.
+	err := msg.BtcDecode(hex.NewDecoder(strings.NewReader(c.Message)),
+		wire.MixVersion)
+	if err != nil {
+		return nil, rpcDeserializationError("Could not decode mix "+
+			"message: %v", err)
+	}
+
+	// Calculate the message hash, so it can be added to known inventory
+	// and used by the sync manager.
+	//
+	// XXX: where does this add the hash to known inventory? (compare to
+	// dcrd/server.go serverPeer.onMixMessage)
+	wsc.blake256HaserMu.Lock()
+	msg.WriteHash(wsc.blake256Hasher)
+	wsc.blake256HaserMu.Unlock()
+	//hash := msg.Hash()
+
+	err = wsc.rpcServer.cfg.SyncMgr.SubmitMixMessage(msg)
```

Comment:
Shouldn't this be returning all of the accepted messages (because orphans can be added) and then relaying them all?

Reply:
Probably.. but the ordering issues aren't so relevant to rpc clients who can/will send them in the proper order.  I don't think I've ever seen dcrd reject a KE from a rpc-mode wallet because the PR was unknown.  I intentionally also increased the time it takes to expire PRs out of dcrd's mixpool, so they would not be removed from underneath the wallet.

Reply:
Leaving open for now. It's not a big issue for now and we can add it later.